### ISSUE TITLE
Improve new player experience

### ISSUE THREAD

**Issue by leemuar** [CleverRaven/Cataclysm-DDA#57571]
### Is your feature request related to a problem? Please describe.

New player experience is harsh. I want to improve that on different levels: UI, quests, tutorial etc

### Solution you would like.

This is an umbrella issue for @silencecoder.
Suggestion on what to improve will be filled here after @silencecoder finish his playtesting research

### Important discussions
https://github.com/CleverRaven/Cataclysm-DDA/issues/57344

Playtests with some some suggestions:
https://www.youtube.com/watch?v=Ys0kGNAAFs0
https://www.youtube.com/watch?v=JlxoRehUgQ0
https://www.youtube.com/watch?v=kIsLdbHUJRw

- [x] figure out how to spawn player on different areas
- [x] figure out how to spawn a player on a specific tile of om_terrain
- [x] implement naive starting location
- [ ] figure out how to show something right after the game starts, without requiring player to make a move
- [ ] figure out how to trigger things tutorial info

**Comment by NetSysFire** [comment]
For the tutorial, see #57285

**Comment by silencecoder** [comment]
## Current Implementation
- Tutorial is hidden in Special menu.
- Tutorial messages start to appear only after a specific key press.
- Tutorial teaches a complex interaction of picking up, reloading and activating a Flashlight.
- Tutorial seems outdated, broken and boring. Player is presented with an artificial environment and inconsequential tasks.
- Play Now! (Fixed Scenario) is meant to be the first session for a new player, which isn't evident from the title.
- NPC in the shelter overwhelms player with information if is persuaded to follow the player. 
- NPC gives a player a random task to complete. Some tasks seems to be easier than others.
- Terminal points the player toward Refugee Center. But an interaction with the terminal is less intuitive than with the NPC, who presumably spend more time in the shelter than player's character.
- In some shelter configurations the player starts in Safe Mode. Thus the player needs to pay attention to the Log and to make a decision about disabling Safe Mode.
- In some shelter configurations there are zombies inside the shelter. Thus the player has to partake in combat or flee from the only known safe zone. 
- Nothing encourages the player to use Crafting. NPC may suggest to board the place up, but doesn't explain how to do this.
- Neither Tutorial nor Play Now! (Fixed Scenario) explains UI to the player.

**Comment by Night-Pryanik** [comment]
> Tutorial is hidden in Special menu.

Though it's not an issue per se, I propose to add additional prompt like "Would you like to start tutorial first?" if player is starting the game for the first time and/or there's no worlds created. As an alternative we could get the tutorial out of Special menu and put it inside a New Game menu. 

> Tutorial messages start to appear only after a specific key press.

Could you please provide examples? Messages are supposed to appear before the actions, not the way around.

> Tutorial seems outdated, broken and boring.

Thanks a lot! Feedback like this really makes me want to continue work on tutorial (no).

 It's only 9 months since I worked on it (#50991), and people are already calling it outdated.

> Player is presented with an artificial environment

Is this bad?

> inconsequential tasks

What's inconsequential with tasks in tutorial?

> Play Now! (Fixed Scenario) is meant to be the first session for a new player

No, it's not. It's meant to start a fixed scenario without hassling with character customization, that's all.

**Comment by leemuar** [comment]
> Thanks a lot! Feedback like this really makes me want to continue work on tutorial (no).
It's only 9 months since I worked on it (https://github.com/CleverRaven/Cataclysm-DDA/pull/50991), and people are already calling it outdated.

@Night-Pryanik please don't take it as a personal offense. 
@silencecoder was very kind to agree to work on improving new player experience as game designer. Currently he is doing playtests and I've asked him to write his impressions here to not forget them and to discuss possible fixes. 
Please note he is playing latest stable (0.F-3) which, as far as I can see, does not have your work from https://github.com/CleverRaven/Cataclysm-DDA/pull/50991 merged. So he is playing that super old and outdated tutorial, thus the "outdated and bad" call

**Comment by silencecoder** [comment]
> It's only 9 months since I worked on it (#50991), and people are already calling it outdated.

The tutorial described in #50991 is **not** the tutorial I and other players played. It seems the described tutorial is not present in the stable build of the game therefore is inaccessible for the most players. I assume it's part of an experimental build.

> Could you please provide examples? Messages are supposed to appear before the actions, not the way around.

The first Tutorial message doesn't show if a player presses Enter, Space and many other buttons. Therefore, the player need to guess that they must Move or Wait in order to display the message.

> > Player is presented with an artificial environment

> Is this bad?

An artificial environment is often mistaken for a safe environment. 

Players need to be reassured about their safety during the initial game experience, and some sort of a training facility is a common shortcut for that. However, this impedes player's decision-making because the training environment doesn't represent actual in-game situations and primes a player for specific choices.

Personally, I prefer to convey a tutorial as a commentary to the actual game. That way the tutorial information helps a player to plan their actions by providing information about the game environment instead of guiding a player through special tasks.

> > Play Now! (Fixed Scenario) is meant to be the first session for a new player

> No, it's not. It's meant to start a fixed scenario without hassling with character customization, that's all.

According to various sources the Evacuee scenario is often recommended to new players as a starting scenario because it doesn't require a character creation and provides relatively safe starting conditions in form of a shelter.

**Comment by Zireael07** [comment]
@silencecoder: "inaccessible to most players because it's in latest stable"... hm, do we have stats of how many people play experimental vs stable?

Also, 0.G is coming soon, maybe this should be postponed until after 0.G, so that everyone is on the same page re the current tutorial?

**Comment by leemuar** [comment]
> hm, do we have stats of how many people play experimental vs stable?

Probably no, but I think this comes from assumption that **new** players download and play release version. The assumption could be wrong, but that's the impression I had too.

@silencecoder is also checking out new tutorial in the experimental branch

**Comment by silencecoder** [comment]
## Target Implementation

Note: This is **not** final and is subject for further improvement.

This tutorial should be implemented as an overlay for the Evacuee scenario. The tutorial information should be conveyed in form of pop-up messages and printed in the Log.

**Part I**. A player starts in a shelter with an NPC.

1. Introduction 
    -  Convey key concepts for the game and the setting
1. UI
    - Explain concept of tiles
    - Briefly explain the sidebar
    - Direct attention toward the Log on the sidebar
1. Movement
    - Explain movement in 8 directions
    - Mention the possibility interaction with character and objects by moving into their tiles (that way we need to explain about doors only after player will open one)
1. Movement+
    - Explain movement between levels
    - Warn about dialogues on moving into various obstacles
    - Briefly explain a concept of speed
1. View
    - Explain how to look around
    - Explain the fog of war concept
    - Explain light levels?
1. View+
    - Explain how to get the list of visible items
    - Explain how to peek (although it's an advanced concept, it connects Move and Look Around actions)
1. Time
    - Explain pause key as a pointer on a tile with the player character
    - Explain Wait actions
    - Explain a duration of in-game actions

> Learned controls:
> - Move
> - Wait
> - Wait for several minutes
> - Descend stairs
> - Ascend stairs
> - Look Around
> - Peek Around Corners
> - Open and Close door (by trigger)

**Part II**. Player should be able to move around the shelter and stumble into a NPC (Non-Player Character). A dialogue with the NPC should be reworked. 
Player shouldn't be able to persuade the NPC to join them, ask what the should do now or ask what they can do for the NPC. Instead the NPC should brief the player on the current situation an d suggest to gather items because the shelter seems to be abandoned.
Player still should be able to trade with the NPC, ask to share an item or for ~tips~ an advice. The latter can be categorized and crowdsourced as actually suggestion from the community to new players. 

1. Interaction 
    - Explain how to examine objects
    - Explain how to smash objects
1. Items
    - Explain how to pick up items
    - Explain how to Wield and Wear picked items
1. Items+
    - Introduce Volume and Weight 
    - Explain characteristics and a state of an item
    - Explain item usage and reload
    - Explain how to throw an item?
1. Inventory 
    - Explain how to handle inventory
    - Explain how to compare items
    - Player can learn other actions for items from the contextual menu

> Learned controls:
>  - Examine nearby terrain 
>  - Smash nearby terrain
>  - Wield
>  - Wear item
>  - Use item
>  - Use wielded item
>  - Open Inventory
>  - Compare two items
>  - Actions with items from a contextual menu (implicitly)

Note: If the player decides to leave the shelter without talking to the NPC or looking for items, there won't be any sequence breaking. Tutorial messages from Part II may be displayed once the player steps on an item.

**Part III**. Player should be able to loot the shelter. After that the NPC may suggest to search the surrounding area and meet at the shelter. NPC should offer the player a choice of a direction and go in the opposite direction. Ideally, the NPC should mention some nearby landmarks since the player character also should know them.

1. Needs
    - Introduce Thirst, Hunger and Sleepiness
    - _Precisely_ explain how often a player need to check on them
    - Explain penalties for thirst, hunger and sleepiness
1. Map
    - ...

Now player should leave shelter and look for useful items nearby. The following tutorial should be triggered once the player steps on a tile outside of the shelter.

1. Movement modes
    - Explain how to switch between Walk, Run, Crouch and Prone
    - Explain the stamina and how to manage breath
    - Explain nose levels
1. Safe mode
    - What is Safe mode and should player disable it
1. Combat
    - ...

**Part IV**. Once player return to the shelter, they should be faced with a choice of leaving the shelter or reinforcing it. 

1. Craft
    - ...
1. Construction
    - ...

---

### Technical Notes 

The tutorial can be implemented as an addition to the existing Play Now! (Fixed Scenario) or as a separate static map.

To avoid the undesirable introduction of Safe Mode or a Combat Encounter it would be nice to be able to clear the immediate vicinity of the shelter from zombies and other hostile entities at the start of the scenario.

Since player character may die in Part III and onward, it's important to preserve some of the player's effort. To do that it would be nice to be able to gradually spawn controllable characters and lead them to the shelter. 
Upon the inevitable player character death a tutorial message should instruct the player on how to proceed in the same world with a new character.

It would be nice if every time there would be a locked door in the shelter.

**Comment by leemuar** [comment]
relates to #57098

**Comment by leemuar** [comment]
NPC can actually teach new players a lot, they have extremely useful information But there is a obstacle: NPC will only share this information if you can "befriend" it first.

![image](https://user-images.githubusercontent.com/5869791/168415431-de4bcc0f-cd35-435e-857a-34e75c0a526d.png)

**Comment by Zireael07** [comment]
> But there is a obstacle: NPC will only share this information if you can "befriend" it first.

I think this makes perfect sense - why would they share the info with someone hostile/stranger?

(That said, a solution might be to make the starting shelter NPC start as friendly?)

**Comment by silencecoder** [comment]
> NPC will only share this information if you can "befriend" it first.
> a solution might be to make the starting shelter NPC start as friendly

I would advice against that for new players. A friendly NPC provides an overwhelming amount of information on demand with no immediate use. In addition to that they teache player how to control NPCs, which is definitely not what new players need at the start of the game.

A friendly NPC with a robust Help system is great as a standalone feature and/or as an organic reminder for game rules. In addition to that even a neutral NPC may provide suggestions to a player crowdsourced from the community.

**Comment by leemuar** [comment]
We can make that starting NPC a special one: it could share a tutorial information, but it will not be a follower right from the start.

**Comment by Zireael07** [comment]
Just to be clear, I did *not* mean a follower, just a friendly (hopefully we can have a friendly NPC w/o it being a follower, which would indeed be overwhelming)

**Comment by leemuar** [comment]
The recording of playtests. Sorry it's in russian, I'm saving it here to have everything related in one place. Maybe Ill add english subtitles later

Please note this in not a critique of the tutorial, it's just a gamedesigner's thoughts on how to improve it. New tutorial is already a huge improvement on what we have previously 

https://studio.youtube.com/video/Ys0kGNAAFs0
https://studio.youtube.com/video/JlxoRehUgQ0
https://studio.youtube.com/video/kIsLdbHUJRw

**Comment by leemuar** [comment]
1st iteration:

[ ] - learn how to set up a predetermined
[ ] - set player to the shelter's basement
[ ] - make first basic lessons (movement, etc) pop-ups

**Comment by leemuar** [comment]
Here are some thoughts after discussing current tutorial design with @silencecoder :

- it is much better to integrate tutorial right in the game rather having a separate game "mode"/"scenario" for it
- "Evacuee" is a nice scenario for beginners in terms of lore and starting dangers/opportunities
- it is better to have a more safe environment at the start to give more time  for the player to figure things out
- shelter's basement looks like a nice place to start: it's a relative safe environment, requires learning how to "go up" the stair, peek stairs/corners, gather first basic supplies, talk to NPC

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/the-game-has-shitty-game-design/28335/1

**Comment by silencecoder** [comment]
> https://discourse.cataclysmdda.org/t/the-game-has-shitty-game-design/28335/1

I'm not sure how this issue is relevant to the discussion above, because the post is a boilerplate complaint about the game not providing a narrative structure to carry a player through the experience. While the goal of this issue is to teach new players about game systems and affordances, which should increase players autonomy in the decision-making process.

**Comment by NetSysFire** [comment]
This was an automated notice. Every time someone mentions an issue or PR in the forums, kevins account is used to relay a notice into here. It may not be relevant and can, as in this case, be disruptive. Most of the time it is fine though.

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/the-game-has-a-shitty-game-design/28335/13

**Comment by leemuar** [comment]
@silencecoder suggested an idea to use "reach evac shelter" quest for beginners

**Comment by silencecoder** [comment]
> @silencecoder suggested an idea to use "reach evac shelter" quest for beginners

To be precise, it was an idea to spawn player in a barricaded place on a boarder of a large city that has an evacuation shelter nearby and then tell a player to reach this shelter. With a pretext that player's character barricaded themself in a safe place but supplies eventually ran out.  

This setup can serve as a small scenario on it's own, because player shouldn't cover too much ground and spend too much time looking for the shelter. We also can conclude the scenarion once player enters the shelter, or add a dynamic "but..." complication:

-  the nearest  shelter is vandalized and abandoned thus a player need to find a new one
-  the nearest  shelter is under the military control thus a player need to obtain a valid ID to enter
- a road to the nearest shelter is blocked thus a player need to sneak around
- the nearest shelter is overcrowded thus a player need to bring their own supplies

That way we can introduce a flimsy replayability and allow player to practice this short adventure multiple times. However, the primary goal is still an addition of a gradual learning curve and a tutorial scaffold to Evacuee scenario.

**Comment by leemuar** [comment]
[#57344 ](https://github.com/CleverRaven/Cataclysm-DDA/issues/57344) has relevant discussion on tutorial messages

**Comment by leemuar** [comment]
not stale, just procrastinating

**Comment by NetSysFire** [comment]
Stalebot is massively malfunctioning right now, flagging and closing issues that should not be closed.

**Comment by leemuar** [comment]
We've finally figured out how to add our custom shelter as a starting location and how to spawn a player at exact coordinates inside.

@silencecoder is working on basic roof layout
My job is to research how popup windows currently work in the game with two goals in mind: 
- show popup message right after scenario is started, without requiring user to press anything
- allow popup message to be opened while the player do stuff

I shared my experience on the game's json loading algo and it helped us to isolate all changes in a separate single file without modifying existing files.
[EvacueeTutorialMVP.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/10475432/EvacueeTutorialMVP.zip)

We've also played Stoneshard for some ideas and "Game Juice" inspiration. We still like the idea to start in the basement

**Comment by Night-Pryanik** [comment]
So, basically you want to throw out all mapgen part of my tutorial rework? Am I understanding you right?

**Comment by NetSysFire** [comment]
I think this might be an addition to the tutorial, so your stuff is still there and this survival/shelter stuff comes after. So you finish the UI basics and can then choose to get teleported to the survival tutorial, where this shelter and map comes into play.

**Comment by leemuar** [comment]
> So, basically you want to throw out all mapgen part of my tutorial rework? Am I understanding you right?

No, we don't want that

**Comment by silencecoder** [comment]
> you finish the UI basics and can then choose to get teleported to the survival tutorial, where this shelter and map comes into play.

The idea of this issue is to provide a tutorial during a normal game session as the player would need it. The game should display relevant pop-ups at the player's pace of in-game exploration. Thus the custom shelter acts just as a test environment for tuning the on-boarding process. 

I'm fully aware that embedding a tutorial structure in a procedurally generated sandbox environment is not a trivial task, so we might end up with a dynamic tutorial system only for Evacuee scenario. Yet the goal is to provide tutorial information as a part of a player generated world.

**Comment by leemuar** [comment]
Digging through how achievements work we've discovered [event_bus ](https://github.com/CleverRaven/Cataclysm-DDA/blob/b991a1bab26d0fb8a6f9068ab4e9c9639ab76e26/src/event_bus.h) by @jbytheway
The general event mechanism could be very useful for the tutorial. We can use it to track different actions and provide meaningful context-based messages to the player.
The first thing we can improve with event_bus is to show a message right after the scenario is started, without requiring player to press anything

**Comment by Brambor** [comment]
This issue looks old and no longer discussed. If not, thank you for your ongoing investigation! Also, this issue is long so I didn't read all of it.

Note: As a developer/tester/designer, you should always play/test the latest experimental. There is no reason not to (other than it is completely broken). Otherwise, you review an old game version and make suggestions based on outdated information. I think you play/test the lastest experimental now (based on https://github.com/CleverRaven/Cataclysm-DDA/issues/57571#issuecomment-1125112228). Just making sure.

**Comment by kevingranade** [comment]
This is ancient,  and also it's a big omnibus issue covering many topics,  which is invalid.
